zubhub_backend/zubhub/activities/models.py

```diff
@@ -0,0 +1,161 @@
+import uuid
+from django.db import models
+from django.contrib.auth import get_user_model
+from django.utils.text import slugify
+from django.utils import timezone
+from math import floor
+
+Creator = get_user_model()
+
+class Category(MP_Node):
+    name = models.CharField(max_length=100, unique=True)
+    description = models.CharField(max_length=1000, blank=True, null=True)
+    slug = models.SlugField(unique=True, max_length=1000)
+    search_vector = SearchVectorField(null=True)
+
+    node_order_by = ['name']
+
+    class Meta:
+        verbose_name_plural = "categories"
+        indexes = (GinIndex(fields=["search_vector"]), )
+
+    def __str__(self):
+        return self.name
+
+    def save(self, *args, **kwargs):
+        if self.slug:
+            pass
+        else:
+            uid = str(uuid.uuid4())
+            uid = uid[0:floor(len(uid) / 6)]
+            self.slug = slugify(self.name) + "-" + uid
+        super().save(*args, **kwargs)
+
+class Image(models.Model):
+    image_url = models.URLField(max_length=1000)
+    public_id = models.CharField(max_length=1000, null=True, blank=True)
+
+    def __str__(self):
+        try:
+            image = self.image_url
+        except AttributeError:
+            image = ''
+        return "Photo <%s:%s>" % (self.public_id, image)    
+
+class InspiringArtist(models.Model):
+    '''this should be having more fields to distinguish an artist '''
+    image = models.ForeignKey(Image,
+                                on_delete=models.CASCADE,
+                                null=True,
+                                related_name="image",
+                                blank=True)
+    short_biography = models.CharField(max_length=10000, null=True, blank=True)
+    name = models.CharField(max_length=100)
+
+    def __str__(self):
+        
+        return self.name
+class Activity(models.Model):
+    id = models.UUIDField(primary_key=True,
+                          default=uuid.uuid4,
+                          editable=False,
+                          unique=True)
+    creators = models.ManyToManyField(Creator,
+                                on_delete=models.CASCADE,
+                                related_name="activities_created")
+    title = models.CharField(max_length=1000)
+    learning_goals = models.CharField(max_length=10000, blank=True, null=True)
+    facilitation_tips = models.CharField(max_length=10000, blank=True, null=True)
```

Comment:
same here, we can use TextField for all places where we need longer length of text.